hist/histpainter/src/THistPainter.cxx

```diff
@@ -4490,6 +4490,10 @@ void THistPainter::Paint(Option_t *option)
       return;
    }
 
+   // Deflate the labels in case of alphanumeric labels
```

Comment:
The correct condition is:
```
if (fXaxis.CanExtend() && fXaxis.IsAlphanumeric()) fH->LabelsDeflate("X");
if (fYaxis.CanExtend() && fYaxis.IsAlphanumeric()) fH->LabelsDeflate("Y");
```

and should we add also the case for the Z axis (for TH3 objects) ?